pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -79,6 +79,18 @@ if [ "$NIX_ENFORCE_PURITY" = 1 -a -n "$NIX_STORE" ]; then
     params=("${rest[@]}")
 fi
 
+if test -n "@libcxx@"; then
+    NIX_CFLAGS_COMPILE="$NIX_CFLAGS_COMPILE -isystem@libcxx@/include/c++/v1"
```

Comment:
@edolstra I think that behavior is different from what we had before, since now every call to your `cc` (not `c++`) results in you getting `-stdlib=libc++` passed in. Before, this was conditional on whether your invocation was to `*++`. Could that lead to bad effects? We could obviously just make up a `NIX_CONDITIONAL_CXX_LDFLAGS` or something and set that in the libc++ setup hook.

I actually like this approach far better, because then we don't even need the skip variables anymore. We just don't have libc++abi depend on libc++abi. Although if we have the stdenv forces us to depend on libc++ then that might get tricky, unless there's an easy way to opt out of build inputs that the stdenv gives you.